prime-dotnet-webapi/Services/ReportingService.cs

```diff
@@ -121,5 +122,39 @@ public async Task<string> UpdatePractitionerTableAsync()
 
             return result;
         }
+
+        public async Task<int> PopulateTransactionLogTempAsync(int numberInDays)
+        {
+            // delete outdated records
+            StringBuilder deleteSql = new StringBuilder();
+            deleteSql.Append("delete from \"PharmanetTransactionLogTemp\"");
+            deleteSql.Append($"where \"TxDateTime\" < current_date + interval '-{numberInDays}' day");
+
+            await _context.Database.ExecuteSqlRawAsync(deleteSql.ToString());
+
+            long? maxTransactionId = await _context.PharmanetTransactionLogTemps.MaxAsync(l => (long?)l.TransactionId);
+
+            // copy records over to temp table
+            StringBuilder copySql = new StringBuilder();
+            copySql.Append("insert into \"PharmanetTransactionLogTemp\"");
+            copySql.Append("(\"Id\", \"CreatedTimeStamp\", \"TransactionId\", \"TxDateTime\", \"UserId\", \"PharmacyId\", \"TransactionType\", ");
+            copySql.Append("\"TransactionSubType\", \"PractitionerId\", \"CollegePrefix\", \"TransactionOutcome\", \"ProviderSoftwareId\", ");
+            copySql.Append("\"ProviderSoftwareVersion\", \"LocationIpAddress\", \"SourceIpAddress\")");
+            copySql.Append("SELECT \"Id\", \"CreatedTimeStamp\", \"TransactionId\", \"TxDateTime\", \"UserId\", \"PharmacyId\", \"TransactionType\", ");
+            copySql.Append("\"TransactionSubType\", \"PractitionerId\", \"CollegePrefix\", \"TransactionOutcome\", \"ProviderSoftwareId\", ");
+            copySql.Append("\"ProviderSoftwareVersion\", \"LocationIpAddress\", \"SourceIpAddress\"");
+            copySql.Append("from \"PharmanetTransactionLog\" ptl ");
+            copySql.Append($"where \"TxDateTime\" > current_date + interval '-{numberInDays}' day ");
+
+            // if there is max transaction Id, use it in the where clause
+            if (maxTransactionId.HasValue)
+            {
+                copySql.Append($"and \"TransactionId\" > '{maxTransactionId}' ");
+            }
```

Comment:
What's the intent of this?  What problem does it avoid/solve?

Reply:
It needs a way to get the only missing records from the original log table - whatever missed from last job run. The best way (I can think of) is by comparing the last/max transaction ID from the temp. However, only if some records already in temp table. If not, skip that step and filter the records by date only.

Reply:
Please add a comment re: the above, I guess for efficiency?
Also does the following help?
```
PTL  t1  XXXXXXXXXXXXXXXXXXXXX00000
     t2  XXXXXXXXXXXXXXXXXXXXXXX1111111

Temp t1                       00000
     t2                         0001111
```

Reply:
Also it means to expand the contents of the PTL Temp table into the past, the PTL Temp table must be cleared first?  I think a comment re: that would be beneficial too

Reply:
More comment added.